pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -213,18 +214,21 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
 
     @Override
     public ProducerBuilder<T> batchingMaxPublishDelay(long batchDelay, @NonNull TimeUnit timeUnit) {
+        checkArgument(batchDelay >= 0, "batchDelay needs to be >= 0");
         conf.setBatchingMaxPublishDelayMicros(timeUnit.toMicros(batchDelay));
         return this;
     }
 
     @Override
     public ProducerBuilder<T> batchingMaxMessages(int batchMessagesMaxMessagesPerBatch) {
+        checkArgument(batchMessagesMaxMessagesPerBatch >= 0, "batchMessagesMaxMessagesPerBatch needs to be >= 0");
         conf.setBatchingMaxMessages(batchMessagesMaxMessagesPerBatch);
         return this;
     }
 
     @Override
     public ProducerBuilder<T> initialSequenceId(long initialSequenceId) {
+        checkArgument(initialSequenceId >= 0, "initialSequenceId needs to be >= 0");
```

Comment:
Why this can’t be negative?

Reply:
There are the similar checks on following flows:
- _deprecated_ `ProducerConfiguration.setBatchingMaxMessages` 
- `ProducerConfiguration.cc` => `setBatchingMaxMessages`